### ISSUE TITLE
Handle readonly numpy arrays

### ISSUE THREAD

**Issue by karlwessel** [hmeine/qimage2ndarray#19]
I'm using this module since three years now and works like a charm, thank you! 

However since numpy 1.17 I get an error when trying to convert readonly arrays to qimage, like in this minimal working example:
```
import numpy as np
import qimage2ndarray as q2n
img = np.zeros((32,32))
q2n.array2qimage(img, normalize=False) # works fine
img.flags['WRITEABLE'] = False
q2n.array2qimage(img, normalize=False)
```
The last line results in the following Exception:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/qimage2ndarray/__init__.py", line 326, in array2qimage
    array = _normalize255(array, normalize)
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/qimage2ndarray/__init__.py", line 204, in _normalize255
    _np.clip(array, low, high, array)
  File "<__array_function__ internals>", line 6, in clip
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 2037, in clip
    return _wrapfunc(a, 'clip', a_min, a_max, out=out, **kwargs)
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 61, in _wrapfunc
    return bound(*args, **kwds)
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/numpy/core/_methods.py", line 132, in _clip
    um.clip, a, min, max, out=out, casting=casting, **kwargs)
  File "/users/kwessel/miniconda36/lib/python3.6/site-packages/numpy/core/_methods.py", line 85, in _clip_dep_invoke_with_casting
    return ufunc(*args, out=out, **kwargs)
ValueError: output array is read-only
```

An easy fix on my side is to just pass a copy of the numpy array to array2qimage. However, since I assume that QImage already needs its own copy of the image data I wonder if it would be possible to convert the image without having to change the input array or create another copy.

**Comment by hmeine** [comment]
Am I right that this is not currently part of the testsuite? I mean – the tests run on your machine, with Numpy 1.7, right? Then, would you be willing to submit a PR introducing a test for this?

**Comment by hmeine** [comment]
Wow… from reading the code now, I get the feeling that you uncovered a bug introduced by me trying to optimize too much: If no normalization is applied, the latest qimage2ndarray would clip the input array to the 0...255 range in-place!

I guess I can quickly introduce a new test based on your code, disregard my question.

**Comment by hmeine** [comment]
Thanks for reporting. I just released a bugfix version 1.8.3.

**Comment by karlwessel** [comment]
Now it works. Thank you for the quick fix!
